bitcoind-rpc/src/main/scala/org/bitcoins/rpc/jsonmodels/RpcPsbtResult.scala

```diff
@@ -0,0 +1,70 @@
+package org.bitcoins.rpc.jsonmodels
+
+import org.bitcoins.core.crypto.{ECDigitalSignature, ECPublicKey}
+import org.bitcoins.core.currency.Bitcoins
+import org.bitcoins.core.protocol.BitcoinAddress
+import org.bitcoins.core.protocol.script.ScriptPubKey
+import org.bitcoins.core.protocol.transaction.Transaction
+import org.bitcoins.core.script.crypto.HashType
+
+sealed abstract class RpcPsbtResult
+
+final case class WalletProcessPsbtResult(psbt: String, complete: Boolean)
+    extends RpcPsbtResult
+
+sealed abstract class FinalizePsbtResult extends RpcPsbtResult
+final case class FinalizedPsbt(hex: Transaction) extends FinalizePsbtResult
+final case class NonFinalizedPsbt(psbt: String) extends FinalizePsbtResult
+
+final case class DecodePsbtResult(
+    tx: RpcTransaction,
+    unknown: Map[String, String],
+    inputs: Vector[RpcPsbtInput],
+    outputs: Vector[RpcPsbtOutput],
+    fee: Option[Bitcoins])
+    extends RpcPsbtResult
+
+final case class RpcPsbtInput(
+    nonWitnessUtxo: Option[RpcTransaction],
+    witnessUtxo: Option[PsbtWitnessUtxoInput],
+    partialSignatures: Option[Map[ECPublicKey, ECDigitalSignature]],
+    sighash: Option[HashType],
+    redeemScript: Option[RpcPsbtScript],
+    witnessScript: Option[RpcPsbtScript],
+    bip32Derivs: Option[Vector[PsbtBIP32Deriv]],
+    finalScriptSig: Option[RpcPsbtScript],
+    finalScriptwitness: Option[Vector[String]], // todo(torkelrogstad)
+    unknown: Option[Map[String, String]] // The unknown global fields
+) extends RpcPsbtResult
+
+final case class RpcPsbtScript(
+    asm: String, // todo(torkelrogstad) split into Vector[ScriptToken]?
+    hex: ScriptPubKey,
+    scriptType: Option[String], // todo(torkelrogstad) found values: witness_v0_keyhash, scripthash, multisig, ???
```

Comment:
Hmm, if we haven't created a ADT for this we probably should. It is used else where in the rpc client

Reply:
I agree. Do you know what the possible values are? The ones in that comment are those I've come across